modules/core/src/parallel.cpp

```diff
@@ -779,8 +773,51 @@ static inline int getNumberOfCPUsImpl()
 }
 #endif
 
+#ifdef __ANDROID__
+static inline int getNumberOfCPUsImpl()
+{
+   FILE* cpuPossible = fopen("/sys/devices/system/cpu/possible", "r");
+   if(!cpuPossible)
+       return 1;
+
+   char buf[2000]; //big enough for 1000 CPUs in worst possible configuration
+   char* pbuf = fgets(buf, sizeof(buf), cpuPossible);
+   fclose(cpuPossible);
+   if(!pbuf)
+      return 1;
+
+   return getNumberOfCPUsImplFromFileStr(pbuf);
+}
+#endif
+
+#if (defined __linux__ || defined __GLIBC__ || defined __HAIKU__ || defined __EMSCRIPTEN__) && !defined __ANDROID__
+static inline unsigned getNumberOfCPUsImplLinux()
+{
+   FILE* cpuPossible = fopen("/sys/fs/cgroup/cpuset/cpuset.cpus", "r");
+   if(!cpuPossible)
+       return 0;
+
+   char buf[2000]; /* big enough for 1000 CPUs in worst possible configuration */
+   char *contents = fgets(buf, sizeof(buf), cpuPossible);
+   fclose(cpuPossible);
+
+   int cpusAvailable = strchr(contents, '-') == NULL ? atoi(contents) : getNumberOfCPUsImplFromFileStr(contents); /* In dockers, this file is a single number, can be (, -) seperated as well*/
```

Comment:
> In dockers, this file is a single number

It can be anything:
```
docker run -it  --cpuset-cpus 1-2,5 ubuntu:18.04
# cat /sys/fs/cgroup/cpuset/cpuset.cpus 
1-2,5
```

Reply:
Yeah, I noticed two types:
1. Just `3` or `4`, etc,  it can be interpreted as CPU 4 or 4 CPUs. If one number it means 4 CPUs
2. Next is `1-2,5`. here I just realized I look only for `-`, I'll look for `,` also, thanks

Reply:
I thought just `3` or `4` means one CPU with this id.

Reply:
Exactly what I thought too, you can try alpine, it gives count, surprised me as well.

Reply:
```
docker run -v $PWD:/var/tmp --cpuset-cpus=3 -it alpine sh
/ # cat /sys/fs/cgroup/cpuset/cpuset.cpus
3
```

```
docker run -v $PWD:/var/tmp --cpuset-cpus=4 -it alpine sh
/ # cat /sys/fs/cgroup/cpuset/cpuset.cpus
4
```

Reply:
What does it show? You set cpuset-cpus and get exactly what you've passed.
```
      --cpus decimal                   Number of CPUs
      --cpuset-cpus string             CPUs in which to allow execution (0-3, 0,1)
```
Docker image does not matter, kernel from the host system is used in containers.

Reply:
Ohh I get what you are saying, ok I thought that was number of CPUs, thanks for the explanation, I'll fix that and the documentation error.